packages/vuelidate/src/core.js

```diff
@@ -23,38 +23,30 @@ import { computed, reactive, ref, watch } from 'vue'
 /**
  * Sorts the validators for a state tree branch
  * @param {Object<NormalizedValidator|Function>} validations
- * @return {{ rules: Object<NormalizedValidator>, nestedValidators: Object, config: Object }}
+ * @return {{ rules: Ref<Object<NormalizedValidator>>, nestedValidators: Ref<Object>, config: Ref<Object> }}
  */
 function sortValidations (validations) {
-  const validationKeys = Object.keys(validations)
-
-  let rules = {}
-  let nestedValidators = {}
-  let config = {}
-
-  validationKeys.forEach(key => {
-    const v = validations[key]
-
-    switch (true) {
-      // If it is already normalized, use it
-      case isFunction(v.$validator):
-        rules[key] = v
-        break
-      // If it is just a function, normalize it first
-      // into { $validator: <Fun> }
-      case isFunction(v):
-        rules[key] = { $validator: v }
-        break
-      // Catch $-prefixed properties as config
-      case key.startsWith('$'):
-        config[key] = v
-        break
-      // If it doesn’t match any of the above,
-      // treat as nestedValidators state property
-      default:
-        nestedValidators[key] = v
-    }
-  })
+  let rules = ref({})
+  let nestedValidators = ref({})
+  let config = ref({})
+
+  watch(validations, (changedValidations) => {
+    // rules.value = {}
+    // nestedValidators.value = {}
+    // config.value = {}
+
+    Object.entries(changedValidations).forEach(([key, v]) => {
+      if (isFunction(v.$validator)) {
+        rules.value[key] = v
+      } else if (isFunction(v)) {
+        rules.value[key] = { $validator: v }
+      } else if (key.startsWith('$')) {
+        config.value[key] = v
+      } else {
+        nestedValidators.value[key] = v
+      }
+    })
+  }, { flush: 'sync', deep: true })
```

Comment:
Why is the `sync` flag needed here? Wouldn’t `deep` be enough?

Reply:
No, because your tree won't be built at first, so you would get errors that $v.number.required is undefined. At least that's what I got.